doc/Architecture/Subtyping.org

```diff
@@ -0,0 +1,161 @@
+* Subtyping
+** An argument about the extent of subtyping
+
+Subtyping is an important part of any module or object system. The
```

Comment:
I don't think Subtyping is a requirement for module/object system. I believe Row Polymorphism is better suited for this task and can be trivially implemented within the dependent typing discipline [citation needed]*.

* not sure about that but I think it's probably true.

Reply:
care describing the technical differences for me? It looks like row polymorphism is like subtyping for records only?

Doe it have the distinction that normal subtyping does not (see the f1, f0, and f2 examples)?

Reply:
The main "problem" with subtyping is that it sometimes makes important information disappear, even for structural subtyping. I believe we can make this work with parametric polymorphism and row polymorphism, and  we can implement algebraic subtyping as a library or something that happens before core.

Reply:
I'd love to see this written up, could you say make a new section about this, and display the arguments and what we could do?

Reply:
@mariari I will try to write it asap, and we can discuss next in the next architecture meeting.